src/test/java/de/tum/in/www1/artemis/JhiMetricsIntegrationTest.java

```diff
@@ -0,0 +1,48 @@
+package de.tum.in.www1.artemis;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+public class JhiMetricsIntegrationTest extends AbstractSpringIntegrationIndependentTest {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    @WithMockUser(username = "admin", roles = "ADMIN")
+    public void getMetricsTest() throws Exception {
+        var result = request.get("/management/jhimetrics", HttpStatus.OK, String.class);
+
+        JsonNode rootNode = objectMapper.readTree(result);
+
+        assertThat(rootNode.has("jvm")).isTrue();
+        assertThat(rootNode.path("jvm").has("G1 Old Gen")).isTrue();
+        assertThat(rootNode.path("jvm").path("G1 Old Gen").has("committed")).isTrue();
+        assertThat(rootNode.path("jvm").path("G1 Old Gen").has("max")).isTrue();
+        assertThat(rootNode.path("jvm").path("G1 Old Gen").has("used")).isTrue();
+
+        assertThat(rootNode.has("http.server.requests")).isTrue();
+        assertThat(rootNode.path("http.server.requests").path("all").has("count")).isTrue();
+
+        assertThat(rootNode.has("cache")).isTrue();
+        assertThat(rootNode.path("cache").path("features").path("cache.size").asDouble()).isEqualTo(8.0);
+
+        assertThat(rootNode.has("garbageCollector")).isTrue();
+        assertThat(rootNode.path("garbageCollector").has("jvm.gc.max.data.size")).isTrue();
+        assertThat(rootNode.path("garbageCollector").path("jvm.gc.pause").has("max")).isTrue();
+
+        assertThat(rootNode.has("processMetrics")).isTrue();
+        assertThat(rootNode.path("processMetrics").path("system.cpu.count").asInt()).isEqualTo(16);
```

Comment:
This test seams to fail

Reply:
ah yes, that check does not really make sense, as it basically tests if the cpu has 16 cores. Seems like the test runners ony have 8, that why it fails. I removed this architecture specific assertion.